vulnerableCode.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2014-2021 Bjoern Kimminich.
+ * SPDX-License-Identifier: MIT
+ */
+
+const utils = require('../lib/utils')
+const insecurity = require('../lib/insecurity')
+const models = require('../models/index')
+const challenges = require('../data/datacache').challenges
+const users = require('../data/datacache').users
+const config = require('config')
+
+module.exports = function login () {
+  function afterLogin (user, res, next) {
+    verifyPostLoginChallenges(user)
+    models.Basket.findOrCreate({ where: { UserId: user.data.id }, defaults: {} })
+      .then(([basket]) => {
+        const token = insecurity.authorize(user)
+        user.bid = basket.id // keep track of original basket for challenge solution check
+        insecurity.authenticatedUsers.put(token, user)
+        res.json({ authentication: { token, bid: basket.id, umail: user.data.email } })
+      }).catch(error => {
+        next(error)
+      })
+  }
+
+  return (req, res, next) => {
+    verifyPreLoginChallenges(req)
+    models.sequelize.query(`SELECT * FROM Users WHERE email = '${req.body.email || ''}' AND password = '${insecurity.hash(req.body.password || '')}' AND deletedAt IS NULL`, { model: models.User, plain: true })
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg" width="20"/></sub>&nbsp; SQL Injection

Unsanitized input from the HTTP request body flows into query, where it is used in an SQL query. This may result in an SQL Injection vulnerability.

<h6>
<a href="https://cwe.mitre.org/data/definitions/89.html">CWE-89</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 769</a> | Line 29</h6>